arcgis-ios-sdk-samples/Route and directions/Navigate route/NavigateRouteViewController.swift

```diff
@@ -159,6 +154,33 @@ class NavigateRouteViewController: UIViewController {
         }
     }
     
+    /// Set route tracker, data source and location display with a solved route result.
+    ///
+    /// - Parameter routeResult: An `AGSRouteResult` object.
+    func setNavigation(with routeResult: AGSRouteResult) {
+        // Set the route tracker
+        routeTracker = makeRouteTracker(result: routeResult)
+        
+        // Set the mock location data source.
+        let firstRoute = routeResult.routes.first!
+        directionsList = firstRoute.directionManeuvers
+        let mockDataSource = makeDataSource(route: firstRoute)
+        initialLocation = mockDataSource.locations?.first
+        // Create a route tracker location data source to snap the location display to the route.
+        let routeTrackerLocationDataSource = AGSRouteTrackerLocationDataSource(routeTracker: routeTracker, locationDataSource: mockDataSource)
+        
+        // Set location display.
+        mapView.locationDisplay.dataSource = routeTrackerLocationDataSource
+        mapView.locationDisplay.autoPanMode = .navigation
+        // If the user navigates the map view away from the location display, activate the recenter button.
+        mapView.locationDisplay.autoPanModeChangedHandler = { [weak self] _ in self?.recenterBarButtonItem.isEnabled = true }
```

Comment:
Instead of setting `recenterBarButtonItem.isEnabled` potentially multiple times, could you remove the handler when it has been called and then re-add it when the recenter button is tapped? Just a thought.

Reply:
I'm not sure if I understood it correctly. 😳  Please refer to the changes on 

https://github.com/Esri/arcgis-runtime-samples-ios/blob/cc37331b0e0e52cc7140e559b053eff3a4871adb/arcgis-ios-sdk-samples/Route%20and%20directions/Navigate%20route/NavigateRouteViewController.swift#L174-L177

and

https://github.com/Esri/arcgis-runtime-samples-ios/blob/cc37331b0e0e52cc7140e559b053eff3a4871adb/arcgis-ios-sdk-samples/Route%20and%20directions/Navigate%20route/NavigateRouteViewController.swift#L226-L229